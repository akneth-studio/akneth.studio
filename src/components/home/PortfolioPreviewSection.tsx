export default function PortfolioPreviewSection() {
  return (
    <section className="portfolio-preview container bg-light py-5 px-4">
      <h2>Portfolio</h2>
      <ul className="list-unstyled">
        <li>
          <strong>Szablon Google Sheets dla kawiarni</strong> - <a href="https://drive.google.com/drive/folders/1NySz1o6OOxVM-yWps897LPPegkeOoqLG?usp=sharing" target="_blank" rel="noopener noreferrer">zobacz na Google Drive</a>
        </li>
        <li>
          <strong>Strona pisarska</strong> - <a href="https://github.com/reisene/wilczar.writer" target="_blank" rel="noopener noreferrer">repozytorium na GitHub</a>
        </li>
        <li>
          <strong>Serwis hulajnóg (prototyp)</strong> - <a href="https://github.com/reisene/HulajDusza-serwis" target="_blank" rel="noopener noreferrer">repozytorium na GitHub</a>
        </li>
      </ul>
      <p className="mt-3">Wkrótce więcej przykładów!</p>
    </section>
  );
}
