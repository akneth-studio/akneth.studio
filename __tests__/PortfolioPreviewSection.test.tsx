import React from 'react';
import { render, screen } from '@testing-library/react';
import PortfolioPreviewSection from '../src/components/home/PortfolioPreviewSection';
import '@testing-library/jest-dom';

describe('PortfolioPreviewSection component', () => {
  it('renders heading and portfolio items with links', () => {
    render(<PortfolioPreviewSection />);
    expect(screen.getByRole('heading', { name: /portfolio/i })).toBeInTheDocument();
    expect(screen.getByText(/szablon google sheets dla kawiarni/i)).toBeInTheDocument();
    expect(screen.getByText(/strona pisarska/i)).toBeInTheDocument();
    expect(screen.getByText(/serwis hulajnóg/i)).toBeInTheDocument();
    expect(screen.getByText(/wkrótce więcej przykładów/i)).toBeInTheDocument();

    // Check links
    expect(screen.getByRole('link', { name: /zobacz na google drive/i })).toHaveAttribute('href');
    const githubLinks = screen.getAllByRole('link', { name: /repozytorium na github/i });
    expect(githubLinks.length).toBeGreaterThan(1);
    githubLinks.forEach(link => {
      expect(link).toHaveAttribute('href');
    });
  });

  it('matches snapshot', () => {
    const { container } = render(<PortfolioPreviewSection />);
    expect(container).toMatchSnapshot();
  });
});
