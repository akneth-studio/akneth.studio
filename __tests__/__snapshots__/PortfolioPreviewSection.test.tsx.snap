// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PortfolioPreviewSection component matches snapshot 1`] = `
<div>
  <section
    class="portfolio-preview container bg-light py-5 px-4"
  >
    <h2>
      Portfolio
    </h2>
    <ul
      class="list-unstyled"
    >
      <li>
        <strong>
          Szablon Google Sheets dla kawiarni
        </strong>
         - 
        <a
          href="https://drive.google.com/drive/folders/1NySz1o6OOxVM-yWps897LPPegkeOoqLG?usp=sharing"
          rel="noopener noreferrer"
          target="_blank"
        >
          zobacz na Google Drive
        </a>
      </li>
      <li>
        <strong>
          Strona pisarska
        </strong>
         - 
        <a
          href="https://github.com/reisene/wilczar.writer"
          rel="noopener noreferrer"
          target="_blank"
        >
          repozytorium na GitHub
        </a>
      </li>
      <li>
        <strong>
          Serwis hulajnóg (prototyp)
        </strong>
         - 
        <a
          href="https://github.com/reisene/HulajDusza-serwis"
          rel="noopener noreferrer"
          target="_blank"
        >
          repozytorium na GitHub
        </a>
      </li>
    </ul>
    <p
      class="mt-3"
    >
      Wkrótce więcej przykładów!
    </p>
  </section>
</div>
`;
